protobuf/model_config.proto

```diff
@@ -1382,6 +1382,24 @@ message ModelSequenceBatching
     //@@     The optional field to specify the initial state for the model.
     //@@
     repeated InitialState initial_state = 5;
+
+    //@@  .. cpp:var:: bool use_single_buffer
+    //@@
+    //@@     The optional field to use a single buffer for both input and output state.
+    //@@     The default value is false.
+    //@@
+    bool use_single_buffer = 6;
+
+    //@@  .. cpp:var:: bool use_growable_memory
+    //@@
+    //@@     The optional field to allow an implicit state buffer to grow or shrink
+    //@@     when the size changes during a sequence. When using this option Triton
+    //@@     guarantess that it will use the same buffer even if the state size changes.
```

Comment:
"guarantees"

Reply:
Does this imply the same as the data content within the changed size is preserved?

Reply:
How about: 

"grow or shrink when the size changes with minimal copies. Triton will reuse existing allocations instead of requesting new memory if possible"

I think "same buffer" is misleading a little here - as its not the "same buffer" if the size has changed?

maybe over thinking the terminology :)

Reply:
> Does this imply the same as the data content within the changed size is preserved?

Yes

Reply:
@nnshah1 I updated the comment. Let me know if it looks better.